tests/src/core/parse.js

```diff
@@ -136,6 +136,6 @@ describe('parse(content, { settings, ecmaFeatures })', function () {
   it('prefers languageOptions.parserOptions over parserOptions', () => {
     const parseSpy = sinon.spy();
     parseStubParser.parse = parseSpy;
-    expect(parse.bind(null, path, content, { settings: {}, parserPath: 'espree', languageOptions: { parserOptions: { sourceType: 'module', ecmaVersion: 2015, ecmaFeatures: { jsx: true } } }, parserOptions: { sourceType: 'script' } })).not.to.throw(Error);
+    expect(parse.bind(null, path, content, { settings: {}, parserPath: 'espree', languageOptions: { sourceType: 'module', ecmaVersion: 2015, parserOptions: { sourceType: 'module', ecmaFeatures: { jsx: true } } }, parserOptions: { sourceType: 'script' } })).not.to.throw(Error);
```

Comment:
this looks to be right based on [the docs](https://eslint.org/docs/latest/use/configure/parser) and my testing:

> In addition to the options specified in `languageOptions.parserOptions`, ESLint also passes `ecmaVersion` and `sourceType` to all parsers. This allows custom parsers to understand the context in which ESLint is evaluating JavaScript code.

It seems that ESLint always populates `parserOptions.sourceType` but not populate `ecmaVersion` based on my testing